### ISSUE TITLE
Ignore url() resolving when bundling CSS

### ISSUE THREAD

**Issue by astashov** [evanw/esbuild#800]
Is there a way to disable resolving the modules inside `url()` in CSS?
I refer to images and fonts in those `url()` like `url(/fonts/font.woff)`, and then copy the fonts to that directory separately. Since I don't have that `/fonts/font.woff` file during the build, the build breaks.

I don't really need the bundler trying to resolve those imports, would be nice to have a way to disable it. Is it possible currently?

Thanks!

**Comment by evanw** [comment]
You should be able to use an on-resolve callback to mark files matching a pattern as external: https://esbuild.github.io/plugins/#resolve-callbacks. You could e.g. mark everything as external that contains `.woff` or that doesn't contain `.css`.

**Comment by astashov** [comment]
Oh nice, it worked!!!

Thank you very much, I just switched my project from Webpack to esbuild, and it reduced the build time from 30 seconds to 2.5 seconds.

esbuild is super awesome, thank you very much for building it!

**Comment by joelmoss** [comment]
I just tried to achieve this using the top level `external` config option, but it doesn't seem to work.

```
{
external: ['/public/fonts/museosansrounded-300-webfont.woff'],
```

**Comment by evanw** [comment]
One additional thing to call out: in the upcoming release (not released yet), `onResolve` plugins will be able to tell what kind of import is being resolved, making it possible to automatically ignore all `url()` imports without needing to be specific about file extensions. You will be able to check for `args.kind === "url-token"`. Details are over at #879.

> Is this supported for CSS `url()` directives?

Yes. That should work. Here's an example:

```
$ echo 'a { background: url(foo.woff) }' | ./esbuild --bundle --loader=css
 > <stdin>:1:16: error: Could not resolve "foo.woff" (mark it as external to exclude it from the bundle)
    1 │ a { background: url(foo.woff) }
      ╵                 ~~~~~~~~~~~~~

1 error
```

```
$ echo 'a { background: url(foo.woff) }' | ./esbuild --bundle --loader=css --external:*.woff
/* <stdin> */
a {
  background: url(foo.woff);
}
```

**Comment by joelmoss** [comment]
Looks like this might be my code. Thanks for verifying, and nice work on the new additions to `onResolve` 👌

**Comment by joelmoss** [comment]
OK, so I figured out the issue.

I have an `onResolve` plugin that filters for paths beginning with a `/`, and resolves this to a new path.

```javascript
build.onResolve({ filter: /^\// }, args => ({
  path: path.join(cwd, args.path)
}))
```

This works great until I set the `external` config option:

```js
esbuild.build({
  external: ['*.woff2']
})
```

My CSS code contains this:

```css
url('/fonts/museosansrounded-100-webfont.woff2')
```

Running esbuild without `external` set will successfully rewrite `/fonts/museosansrounded-100-webfont.woff2` to `/Users/joelmoss/dev/froxy/test/internal/fonts/museosansrounded-100-webfont.woff2`.

But running esbuild with `external` set raises:

```
 > lib/reset.css:10:38: error: Could not read from file: /Users/joelmoss/dev/froxy/test/internal/fonts/museosansrounded-100-webfont.woff2
    10 │   url('/fonts/museosansrounded-100-webfont.woff2') 
```

So what I think is happening here, is that the `onResolve` plugin is executed before the `external` config option is applied, resulting in weird goings on.

Shouldn't the `external` option be applied before plugins, so that they don't attempt to resolve/load externals?
